### ISSUE TITLE
Power, Z, ARM requires support for dynamicANewArray?

### ISSUE THREAD

**Issue by 0xdaryl** [eclipse/openj9#6441]
There is a feature in VP guarded by a code generator flag `SupportsDynamicANewArray`.  The comment in VP claims support is not yet implemented on platforms other than x86, hence this transformation will not occur:

https://github.com/eclipse/openj9/blob/582921ee5d7d99a01cdf1b98f0dfa92cad3a72cd/runtime/compiler/optimizer/J9ValuePropagation.cpp#L1419

This comment pre-dates open sourcing.  Does anyone recall what this is about, and whether there is some work to do on P, Z, and ARM?

@andrewcraik @gita-omr @fjeremic @knn-k

**Comment by andrewcraik** [comment]
@0xdaryl Yes there was an opt done by @yanluo7 in OMR VP and OpenJ9 to accelerate array cloning:
https://github.com/eclipse/omr/pull/3421
https://github.com/eclipse/openj9/pull/4179

The evaluator for `anewarray` needs to support an array type that is a non-constant `loadaddr`. It isn't a big change, but that is needed for the acceleration done in those pull requests. I was sure an issue was opened but I can't seem to find it now...

**Comment by fjeremic** [comment]
I looked on my personal tracking repository for issues that need to be addressed on Z and I can't find the relevant issue either.

**Comment by 0xdaryl** [comment]
This will be the issue to track enabling support for dynamicANewArray on all remaining platforms.  Rather than a contribution from each platform, I'd like to have this support added and enabled on all remaining architectures at once (sharing code as appropriate).

I'll change the title of this issue to reflect the new scope.
